File: c#/program41.cs
using System;

class Pattern
{
    public int iRow;
    public int iCol;

    public Pattern(int no1,int no2)
    {
        this.iRow = no1;
        this.iCol = no2;
    }

    public void DisplayPattern()
    {
        int i = 0,j = 0;

        if(iRow != iCol)
        {
            Console.WriteLine("Invalid Input");
            return;
        }
        
        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i <= j)
                {
                    Console.Write("*\t");
                }
                else
                {
                    Console.Write(" \t");
                }
            }
            Console.WriteLine();
        }
        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i <= j)
                {
                    Console.Write("*\t");
                }
                else
                {
                    Console.Write(" \t");
                }  
            }
            Console.WriteLine();
        }
        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                if(i <= j)
                {
                    Console.Write("*\t");
                }
                else
                {
                    Console.Write(" \t");
                }  
            }
            Console.WriteLine();
        }
    }
}

class MainClass
{
    public static void Main()
    {
        int value1 = 0,value2 = 0;

        Console.WriteLine("Enter number of rows :");
        value1 = int.Parse(Console.ReadLine());

        Console.WriteLine("Enter number of columns :");
        value2 = int.Parse(Console.ReadLine());

        Pattern obj = new Pattern(value1,value2);
        obj.DisplayPattern();
    }
}

File: cpp/program41.cpp
// program150.cpp

#include<iostream>
using namespace std;

class Pattern
{
    public:
        int iRow;
        int iCol;

    Pattern(int X, int Y)
    {
        iRow = X;
        iCol = Y;
    }

    void DisplayPattern()
    {
        int i = 0;
        int j = 0;

        if(iRow != iCol)
        {
            cout<<"Invalid Input"<<"\n";
            return;
        }

        for(i = 1; i < iRow; i++)
        {
            for(j = 1; j <= i; j++)
            {
                if(i >= j)
                {
                    cout<<"*"<<"\t";
                }
            }
            cout<<"\n";
        }

        for(i = iRow; i >= 1; i--)
        {
            for(j = 1; j <= i; j++)
            {
                cout<<"*"<<"\t";  
            }
            cout<<"\n";
        }
    }
};

int main()
{
    int iValue1 = 0;
    int iValue2 = 0;

    cout<<"Enter number of Rows"<<"\n";
    cin>>iValue1;

    cout<<"Enter number of Columns"<<'\n';
    cin>>iValue2;

    Pattern pobj(iValue1,iValue2);
    pobj.DisplayPattern();

    return 0;
}